src/dwarf_parser.cpp

```diff
@@ -0,0 +1,164 @@
+#include "dwarf_parser.h"
+
+#ifdef HAVE_LIBDW
+
+#include "log.h"
+
+#include <dwarf.h>
+#include <elfutils/libdw.h>
+
+namespace bpftrace {
+
+struct FuncInfo
+{
+  std::string name;
+  Dwarf_Die die;
+};
+
+Dwarf::Dwarf(const std::string &file_path)
+{
+  callbacks.find_debuginfo = dwfl_standard_find_debuginfo;
+  callbacks.section_address = dwfl_offline_section_address;
+  dwfl = dwfl_begin(&callbacks);
+  if (dwfl_report_offline(dwfl, file_path.c_str(), file_path.c_str(), -1) ==
+      NULL)
+  {
+    LOG(WARNING) << "Could not read DWARF info from " + file_path;
+  }
+  dwfl_report_end(dwfl, NULL, NULL);
+}
+
+std::unique_ptr<Dwarf> Dwarf::GetFromBinary(const std::string &file_path)
+{
+  std::unique_ptr<Dwarf> dwarf(new Dwarf(file_path));
```

Comment:
Nit: more idiomatic and cuts down memory allocations from 2 -> 1 (not that we really care about perf here)
```suggestion
  auto dwarf = std::make_unique<Dwarf>(file_path);
```

Reply:
Unfortunately, `make_unique` cannot access the private constructor from a static method.

Reply:
Ah too bad